### PR TITLE
JNG-4317 add deep copy tests

### PR DESCRIPTION
**Pull request by Madmat8**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4317" title="JNG-4317" target="_blank">TaskJNG-4317</a>  Support deep copy for composition
  </td></table>
  <br />

### CODE REVIEW

File: judo-runtime-core-jsl-itest/models/TransferAggregationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAssociationAggregationTest.java [hunk 1]
```diff
@@ -375,6 +381,7 @@ public void testSingleRequiredAssociationAggregationRelationOnTransfer() {
      */
     @Test
     @TestCase("MultiAssociationAggregationRelationOnTransfer")
+    @Disabled
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/TransferAggregationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAssociationAggregationTest.java]: Is this intentional? If so, maybe we should add a ticket as note?

File: judo-runtime-core-jsl-itest/models/TransferAssociationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferCompositonAssociationTest.java [hunk 2]
```diff
@@ -124,13 +122,12 @@ public void testSingleCompositionAssociationOnTransfer() {
                 TransferAAttachedRelationsForCreate.builder().withSingleEntityB(transferB).build()
         );
 
-        // TODO JNG-4317 Composition should copy the composition element or not allow to bind an existed element.
-//        assertEquals(2,transferBDao.query().execute().size());
-//        assertEquals(2,entityBDao.query().execute().size());
+        assertEquals(2,transferBDao.query().execute().size());
+        assertEquals(2,entityBDao.query().execute().size());
 
         // Check transferA contains transferB
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/TransferAssociationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferCompositonAssociationTest.java:135]: Do we mean "does not"?

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java [hunk 3]
```diff
@@ -473,4 +482,60 @@ void testCompositionWithUndefinedRequiredFields() {
...
+        assertEquals(11, entityCDao.countAll());
+        assertEquals(9, entityDDao.countAll());
+
+        a3.setSingleConA(EntityC.builder().withStringC("C3Updated").build());
+        a3.setSingleRequiredConA(EntityC.builder().withStringC("C4Updated").withMultipleDonB(List.of(EntityD.builder().withStringD("D3Updated").build())).build());
+
+        entityADao.update(a3);
+
+        assertEquals(11, entityCDao.countAll());
+        // TODO: JNG-5213 update does not create new EntityD instance
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java:535]: I tested nested composition update and create and both work if payloads are prepared properly. TBD

**Review by noherczeg — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Review by gaborflorian — APPROVED** [review]

File: judo-runtime-core-jsl-itest/models/TransferAggregationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAssociationAggregationTest.java (hunk 1 above)
Comment (robertcsakany) [reply judo-runtime-core-jsl-itest/models/TransferAggregationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAssociationAggregationTest.java]: Please give ticket number for disabled test.

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/TransferAggregationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAssociationAggregationTest.java (hunk 1 above)
Comment (Madmat8) [reply judo-runtime-core-jsl-itest/models/TransferAggregationModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAssociationAggregationTest.java]: [210fa36](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/244/commits/210fa3623369ae45e685aeb21cee5392ea0181c6)

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**